server/src/main/kotlin/org/elaastic/player/PlayerController.kt

```diff
@@ -761,44 +715,83 @@ class PlayerController(
     }
 
     /**
-     * Get the LearnerResultsModel for the given responses and statement
+     * Get the result modal for the given learner
      *
-     * @param responseFirstAttempt the response of the learner for the first attempt
-     * @param responseSecondAttempt the response of the learner for the second attempt
-     * @param statement the statement of the sequence
+     * @param authentication the current authentication
+     * @param model the model
+     * @param sequenceId the id of the sequence to get the result modal
+     * @param userId the id of the learner to get the result modal
+     * @return the result modal for the given learner
      */
-    private fun builtLearnerResultsModel(
-        responseFirstAttempt: Response?,
-        responseSecondAttempt: Response?,
-        responseFirstTryHasChatGPTEvaluation: Boolean,
-        responseSecondTryHasChatGPTEvaluation: Boolean,
-        statement: Statement
+    @GetMapping("/{sequenceId}/result/{userId}/modal")
+    fun resultModal(
+        authentication: Authentication,
+        model: Model,
+        @PathVariable sequenceId: Long,
+        @PathVariable userId: Long
+    ): String {
+        val user: User = authentication.principal as User
+
+        val sequence = sequenceService.get(sequenceId, true)
+        val learner = userService.findById(userId)
```

Comment:
I wonder why this is `findById` and not `get` here... I have to check this.
Usually `get` is a _find by id_ using the cache...

Reply:
I've checked `UserService`. The both methods are defined : `get()` & `findById` with a different logic. 
If the two methods are required, probably that they should reference each other and the difference should be clarified in the doc.

I remember on a previous framework both `get(...)` and `findById(...)` was automatically defined, the difference being that `get` was using the cache.

I've seen on Elaastic that on `UserRepository` I have implemented `findById` as a search, possible returning `null` and `get` as a user retrieval, throwing an exception when not defined.

Reply:
It's the opposite, `get` return `User?` and `findById` throw an exception if the user is not find : 
```kotlin
@Throws(IllegalArgumentException::class)
fun findById(learnerId: Long): User {
    return userRepository.findById(learnerId).orElseThrow { IllegalArgumentException("Learner not found") }
}
```

Reply:
Well... I'm not sure of what I've read anymore... In `UserRepository` the `findById` method has an optional result, but there is not `get` method.

It is a matter of convention. It is more natural to me to use `find` for a search with can have no result whereas `get` is to get an expected result.

Reply:
We could mark that as an issue in GitHub and I could fix it later.

Reply:
I will do it